### PR TITLE
Web Services JDK8 Utilities and Driver unit test updates

### PR DESCRIPTION
**Pull request by jonreding2010**
Wrote unit tests for Get, Delete, Post, Put, and Patch
Made annotation in Product class to get xml to work and to add a namespace in the xml during the test

### CODE REVIEW

File: jmaqs-webservices-jdk8/src/test/java/com/magenic/jmaqs/webservices/jdk8/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+   */
+  @Test(groups = TestCategories.WEB_SERVICE)
+  public void webServiceStringPut() throws Exception {
+    Product p = new Product();
+    p.setCategory("ff");
+    p.setId(4);
+    p.setName("ff");
+    p.setPrice(BigDecimal.valueOf(3.25));
+
+    HttpEntity content = WebServiceUtilities.createStringEntity(p, ContentType.APPLICATION_JSON);
```
Comment (bradyrad) [inline jmaqs-webservices-jdk8/src/test/java/com/magenic/jmaqs/webservices/jdk8/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java:70]: Should the Content Type be TEXT_PLAIN here as the test is supposed to serialize as string?

Comment (jonreding2010) [reply jmaqs-webservices-jdk8/src/test/java/com/magenic/jmaqs/webservices/jdk8/WebServiceDriverRequestUnitTest/WebServiceDriverPutUnitTest.java]: TEXT_PLAIN has replaced ContentType.APPLICATION_JSON

**Review by bradyrad — APPROVED** [review]
All necessary changes resolved.

File: jmaqs-webservices-jdk8/src/test/java/com/magenic/jmaqs/webservices/jdk8/WebServiceDriverRequestUnitTest/WebServiceDriverPatchUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+   * Verify a patch error returns the expected code and message.
+   *
+   * @throws Exception
+   *           There was a problem with the test
+   */
+  @Test(groups = TestCategories.WEB_SERVICE)
+  public void webServicePatchError() throws Exception {
+    HttpEntity content = WebServiceUtilities.createStringEntity("", ContentType.APPLICATION_XML);
+    CloseableHttpResponse response = this.getWebServiceDriver()
+        .patchContent("/api/XML_JSON/Put/1", content, ContentType.APPLICATION_XML, false);
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk8/src/test/java/com/magenic/jmaqs/webservices/jdk8/WebServiceDriverRequestUnitTest/WebServiceDriverPatchUnitTest.java]: put has been changed to patch

File: jmaqs-webservices-jdk8/src/main/java/com/magenic/jmaqs/webservices/jdk8/WebServiceUtilities.java [hunk 3]
```diff
@@ -124,6 +138,8 @@ public static String getResponseBody(CloseableHttpResponse response) throws IOEx
    * @throws JsonProcessingException the json processing exception
    */
   public static <T> String serializeXml(T body) throws JsonProcessingException {
+    xmlMapper.configure(ToXmlGenerator.Feature.WRITE_XML_DECLARATION, true);
+    xmlMapper.setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk8/src/main/java/com/magenic/jmaqs/webservices/jdk8/WebServiceUtilities.java]: Deleted setSerializationInclusion and moved it into the Product class

File: jmaqs-webservices-jdk8/src/main/java/com/magenic/jmaqs/webservices/jdk8/WebServiceUtilities.java [hunk 4]
```diff
@@ -92,11 +103,14 @@ public static String getResponseBody(CloseableHttpResponse response) throws IOEx
...
+      throws JsonProcessingException, UnsupportedEncodingException {
     if (contentType.toString().toUpperCase().contains("XML")) {
       return new StringEntity(serializeXml(body), contentType);
     } else if (contentType.toString().toUpperCase().contains("JSON")) {
       return new StringEntity(serializeJson(body), contentType);
+    } else if (contentType.toString().toUpperCase().contains("TEXT")) {
+      return new StringEntity((String) body);
     } else {
       throw new IllegalArgumentException(
           StringProcessor.safeFormatter("Only xml and json conversions are currently supported"));
```
[the remark this answers is not in the document]
Comment (jonreding2010) [reply jmaqs-webservices-jdk8/src/main/java/com/magenic/jmaqs/webservices/jdk8/WebServiceUtilities.java]: updated message string and updated the return for "TEXT" content type

**Pull request merged**